hyperspy/misc/machine_learning/import_sklearn.py

```diff
@@ -27,3 +27,5 @@
         sklearn_installed = True
 except ImportError:
     sklearn_installed = False
+finally:
+    warnings.simplefilter("default")
```

Comment:
This should also be redundant as context managers get released even if an exception happens in the `with` block.